server/src/main/java/io/deephaven/server/console/NoConsoleSessionModule.java

```diff
@@ -28,8 +28,7 @@ ScriptSession bindScriptSession(NoLanguageDeephavenSession noLanguageSession) {
     @Provides
     NoLanguageDeephavenSession bindNoLanguageSession(
             @Named(PeriodicUpdateGraph.DEFAULT_UPDATE_GRAPH_NAME) final UpdateGraph updateGraph,
-            final OperationInitializer operationInitializer,
-            final ScriptSessionCacheInit ignored) {
+            final OperationInitializer operationInitializer) {
```

Comment:
This doesn't seem like exactly the right change. The issue appears to be with parallel execution, where one test creates the directory before another gets around to it.

Reply:
```
Index: engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java b/engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java
--- a/engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java	(revision 25e0cb1ff86be7d7814085f8e096e9db4661f4bd)
+++ b/engine/table/src/main/java/io/deephaven/engine/util/AbstractScriptSession.java	(date 1715205168607)
@@ -26,6 +26,7 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.*;
 import java.util.function.Function;
@@ -41,26 +42,27 @@
 
     private static final Path CLASS_CACHE_LOCATION = CacheDir.get().resolve("script-session-classes");
 
-    protected static File newClassCacheLocation() {
+    protected static Path newClassCacheLocation() {
         // TODO(deephaven-core#1713): Introduce instance-id concept
         final UUID scriptCacheId = UuidCreator.getRandomBased();
-        final File directory = CLASS_CACHE_LOCATION.resolve(UuidCreator.toString(scriptCacheId)).toFile();
+        final Path directory = CLASS_CACHE_LOCATION.resolve(UuidCreator.toString(scriptCacheId));
         createOrClearDirectory(directory);
         return directory;
     }
 
     public static void createScriptCache() {
-        final File classCacheDirectory = CLASS_CACHE_LOCATION.toFile();
-        createOrClearDirectory(classCacheDirectory);
+        createOrClearDirectory(CLASS_CACHE_LOCATION);
     }
 
-    private static void createOrClearDirectory(final File directory) {
-        if (directory.exists()) {
-            FileUtils.deleteRecursively(directory);
+    private static void createOrClearDirectory(final Path directory) {
+        if (Files.exists(directory)) {
+            FileUtils.deleteRecursively(directory.toFile());
         }
-        if (!directory.mkdirs()) {
+        try {
+            Files.createDirectories(directory);
+        } catch (IOException e) {
             throw new UncheckedDeephavenException(
-                    "Failed to create class cache directory " + directory.getAbsolutePath());
+                    "Failed to create class cache directory " + directory.toAbsolutePath(), e);
         }
     }
 
@@ -78,7 +80,7 @@
             final OperationInitializer operationInitializer,
             final ObjectTypeLookup objectTypeLookup,
             @Nullable final Listener changeListener) {
-        this(updateGraph, operationInitializer, objectTypeLookup, changeListener, newClassCacheLocation(),
+        this(updateGraph, operationInitializer, objectTypeLookup, changeListener, newClassCacheLocation().toFile(),
                 Thread.currentThread().getContextClassLoader());
     }
 
```